src/Microsoft.ML.Dnn/DnnRetrainTransform.cs

```diff
@@ -857,7 +861,7 @@ protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, b
                 return Utils.MarshalInvoke(MakeGetter<int>, type, input, iinfo, srcTensorGetters, activeOutputColNames, outputCache);
             }
 
-            private Delegate MakeGetter<T>(DataViewRow input, int iinfo, ITensorValueGetter[] srcTensorGetters, string[] activeOutputColNames, OutputCache outputCache)
+            private Delegate MakeGetter<T>(DataViewRow input, int iinfo, ITensorValueGetter[] srcTensorGetters, string[] activeOutputColNames, OutputCache outputCache) where T: unmanaged
```

Comment:
`where T: unmanaged`

Can this ever be used with a `string` type? If so, this is going to fail at runtime since we use reflection to call this method.

Reply:
I have made a separate function for when T is string to avoid this failure.

---
In reply to: [324317321](https://github.com/dotnet/machinelearning/pull/4205#discussion_r324317321) [](ancestors = 324317321)